File: Dtos/PriceAlertDto.cs
using api.Models;

namespace api.Dtos
{
    public class PriceAlertDto
    {
        public int Id { get; set; }
        public int StockId { get; set; }
        public string StockSymbol { get; set; } = string.Empty;
        public string CompanyName { get; set; } = string.Empty;
        public decimal TargetPrice { get; set; }
        public AlertType AlertType { get; set; }
        public bool IsActive { get; set; }
        public DateTime CreatedAt { get; set; }
        public DateTime? TriggeredAt { get; set; }
    }
}

File: src/main/java/com/stockwatch/stockwatchpro/dtos/PriceAlertDto.java
package com.stockwatch.stockwatchpro.dtos;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;
import java.time.LocalDateTime;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class PriceAlertDto {
    private Integer id;
    private String appUserId;
    private Integer stockId;
    private BigDecimal targetPrice;
    private String alertType;
    private Boolean isActive;
    private LocalDateTime createdAt;
    private LocalDateTime triggeredAt;
}
